codechain_primitives/UnsignedInteger.py

```diff
@@ -0,0 +1,140 @@
+
+    def times(self, rhsValue):
+        return self.__class__(int(self) * int(rhsValue))
+
+    def idiv(self, rhsValue):
+        if int(rhsValue) < 0:
+            raise ValueError("Negative division")
+        return self.__class__(int(self) // int(rhsValue))
+
+    def mod(self, rhsValue):
+        if int(rhsValue) < 0:
+            raise ValueError("Negative modulo")
+        return self.__class__(int(self) % int(rhsValue))
+
+    def __add__(self, rhsValue):
+        return self.plus(rhsValue)
+
+    def __sub__(self, rhsValue):
+        return self.minus(rhsValue)
+
+    def __mul__(self, rhsValue):
+        return self.times(rhsValue)
+
+    def __floordiv__(self, rhsValue):
+        return self.idiv(rhsValue)
+
+    def __mod__(self, rhsValue):
+        return self.mod(rhsValue)
+
+    @classmethod
+    def from_rlp(cls, buffer):
+        if type(buffer) is not bytes and type(buffer) is not bytearray:
+            raise ValueError("Argument should be bytearray")
+        data = buffer[1:]
+        first = buffer[0]
+        if first < 0x80:
+            return cls(hex(first))
+
+        length = first - 0x80
+        if len(data) != length:
+            raise ValueError("Invalid data for Unsigned integer")
+        elif length > len(bytes.fromhex(hex(cls.MAX_VALUE)[2:])):
+            raise ValueError(
+                f"Data for Unsigned integer must be less than or equal to {len(bytes.fromhex(hex(cls.MAX_VALUE)[2:]))}")
+        elif length == 0:
+            return cls('0')
+        return cls(int.from_bytes(data, byteorder='big'))
+
+    @classmethod
+    def check_string(cls, param):
+        if type(param) is not str:
+            return False
+        try:
+            value = int(param, 0)
+            return value >= 0 and value <= cls.MAX_VALUE
+        except:
+            return False
+
+    @classmethod
+    def check(cls, param):
+        if type(param) is str:
+            return cls.check_string(param)
+        else:
+            if not isinstance(param, int):
+                return False
+        if param >= 0 and param <= cls.MAX_VALUE:
+            return True
+        return False
+
+    def to_encode_object(self):
+        result = hex(int(self))[2:]
+        result = "0" + result if len(result) % 2 == 1 else result
+        return bytes.fromhex(result)
+
+    def rlp_bytes(self):
+        return encode(int(self))
+
+    def is_equal_to(self, rhs):
+        return int(self) == int(rhs)
+
+    def is_greater_than(self, rhs):
+        return int(self) > int(rhs)
+
+    def is_greater_than_or_equal_to(self, rhs):
+        return int(self) >= int(rhs)
+
+    def is_less_than(self, rhs):
+        return int(self) < int(rhs)
+
+    def is_less_than_or_equal_to(self, rhs):
+        return int(self) <= int(rhs)
+
+    def to_string(self, base=16, prefix=True):
+        return str(int(self)) if base == 10 else (hex(int(self))[2:] if prefix == False else hex(int(self)))

```

Comment:
Spliting it to multi line would be more readible. Use format string rather than string slicing.
```python
if base == 10:
	return str(int(self))
elif base == 16 and prefix:
	return hex(int(self))
elif base == 16 and not prefix:
	return "{:x}".format(self)
else
	raise ValueError("Only supports base 10, 16")
```
Implement `__str__` too.

Reply:
I think we can use `__str__` of int class.